validateWorkerVersions.ps1

```
@@ -0,0 +1,87 @@
+        throw "Failed to find version for package $packageName"
+    }
+    return $version
+}
+
+function setCliPackageVersion([string]$packageName, [string]$version)
+{
+    $node = $cliCsprojXml.SelectSingleNode("/Project//PackageReference[@Include='$packageName']")
+    $node.Version = $version
+}
+
+$hostPackageName = "Microsoft.Azure.WebJobs.Script.WebHost"
+if (-Not $hostVersion) {
+    $hostVersion = getPackageVersion $hostPackageName $cliCsprojContent
+} elseif ($Update) {
+    setCliPackageVersion $hostPackageName $hostVersion
+}
+
+$hostRepoUrlBase = "https://raw.githubusercontent.com/Azure/azure-functions-host/v$hostVersion"
+$hostCsprojContent = removeBomIfExists((Invoke-WebRequest -Uri "$hostRepoUrlBase/src/WebJobs.Script/WebJobs.Script.csproj").Content)
+$pythonPropsContent = removeBomIfExists((Invoke-WebRequest -Uri "$hostRepoUrlBase/build/python.props").Content)
+
+$workers = "JavaWorker", "NodeJsWorker", "PowerShellWorker.PS7.0", "PowerShellWorker.PS7.2", "PythonWorker"
+
+$failedValidation = $false
+foreach($worker in $workers) {
+    $packageName = "Microsoft.Azure.Functions.$worker"
+    if ($worker -eq "PythonWorker") {
+        $hostWorkerVersion = getPackageVersion $packageName $pythonPropsContent
+    } else {
+        $hostWorkerVersion = getPackageVersion $packageName $hostCsprojContent
+    }
+    $cliWorkerVersion = getPackageVersion $packageName $cliCsprojContent
+
+    if ($Update) {
+        setCliPackageVersion $packageName $hostWorkerVersion
+    } elseif ($hostWorkerVersion -ne $cliWorkerVersion) {
+        Write-Output "Reference to $worker in the host ($hostWorkerVersion) does not match version in the cli ($cliWorkerVersion)"
+        $failedValidation = $true
+    }
+}
+
+if ($Update) {
+    $cliCsprojXml.Save($cliCsprojPath)
+    Write-Output "Updated worker versions! 🚀"
+} elseif ($failedValidation) {
+    Write-Output "You can run './validateWorkerVersions.ps1 -Update' locally to fix worker versions."

```

Ah, I see. This runs with every build pipeline, so the pipeline will fail early. Running the script locally will apply the reference changes.

---

Thanks @ejizba for this.

For fixing worker versions, can we create a GitHub action which - given a host version - fixes the worker versions (sends in a PR for the fix)?

---

Yes, I think it be would a logical next-step to automate this in some way